systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaOauthExampleClients.java

```diff
@@ -8,52 +8,138 @@
 import io.strimzi.api.kafka.model.KafkaResources;
 import io.strimzi.systemtest.keycloak.KeycloakInstance;
 
-public class KafkaOauthClientsResource extends KafkaBasicClientResource {
+import java.security.InvalidParameterException;
+
+public class KafkaOauthExampleClients extends KafkaBasicExampleClients {
 
     private final String oauthClientId;
     private final String oauthClientSecret;
     private final String oauthTokenEndpointUri;
     private final String userName;
 
-    public KafkaOauthClientsResource(
-        String producerName, String consumerName, String bootstrapServer, String topicName, int messageCount,
-        String additionalConfig, String consumerGroup, String oauthClientId, String oauthClientSecret, String oauthTokenEndpointUri) {
+    public static class Builder extends KafkaBasicExampleClients.Builder {
+        private String oauthClientId;
+        private String oauthClientSecret;
+        private String oauthTokenEndpointUri;
+        private String userName;
+
+        public Builder withOAuthClientId(String oauthClientId) {
+            this.oauthClientId = oauthClientId;
+            return self();
+        }
+
+        public Builder withOAuthClientSecret(String oauthClientSecret) {
+            this.oauthClientSecret = oauthClientSecret;
+            return self();
+        }
+
+        public Builder withOAuthTokenEndpointUri(String oauthTokenEndpointUri) {
+            this.oauthTokenEndpointUri = oauthTokenEndpointUri;
+            return self();
+        }
+
+        public Builder withUserName(String userName) {
+            this.userName = userName;
+            return self();
+        }
+
+        @Override
+        public Builder withProducerName(String producerName) {
+            return (Builder) super.withProducerName(producerName);
+        }
```

Comment:
Here you are mixing two things. The explicit casting of the Builder and using my implementation `self()`, which encapsulate this casting.

Reply:
Well I removed the `self()` method and keep only `return this` in cases which are specific -> for Oauth example client is `withOAuthClientId..` etc.

For the method that I'm overriding I keep the `return (Builder) super. ..` as we want to use the methods from the `KafkaBasicExampleClients`. If I do it with `return self()` or `return this` here, the methods will not work -> as they don't have access on it.

Can it be like that?